### PR TITLE
implement request_to_curl function

### PR DESCRIPTION
**Pull request by guillermo-bondonno**
Implement request_to_curl util function that generates a curl command string from a scrapy.Request object

### CODE REVIEW

File: scrapy/utils/request.py [hunk 1]
```diff
@@ -327,3 +327,35 @@ def _get_method(obj, name):
...
+        "-H '{}: {}'".format(k.decode(), v[0].decode())
+        for k, v in request.headers.items()
+    )
+    url = request.url
+    cookies = ""
+    if isinstance(request.cookies, dict) and request.cookies:
+        cookies = "--cookie '{}'".format(
+            "; ".join("{}={}".format(k, v) for k, v in request.cookies.items())
+        )
+    elif isinstance(request.cookies, list):
```
Comment (wRAR) [inline scrapy/utils/request.py:353]: Maybe this also needs a check that the list is not empty?

**Comment by wRAR** [comment]
Can you please also look into pylint failures?

File: scrapy/utils/request.py (hunk 1 above)
Comment (guillermo-bondonno) [reply scrapy/utils/request.py:353]: Makes sense. The default is an empty dict, but I guess it is possible that an empty list ends up there.

**Comment by guillermo-bondonno** [comment]
> Can you please also look into pylint failures?

@wRAR It's about using .format instead of f-strings. I think it's much cleaner to use format in this case instead of doing something like

`cookies = f"--cookie '{'; '.join(f'{list(c.keys())[0]}={list(c.values())[0]}' for c in request.cookies)}'"`

but if it can't be merged as it is, I'll find a way around it sure.
What do you think?

**Comment by wRAR** [comment]
I think this warning should indeed be disabled, per-line or globally. I'll wait for other opinions though.

**Review by wRAR — APPROVED** [review]

**Review by Gallaecio — APPROVED** [review]
I think it would be ideal to cover this somewhere in the docs for discoverability, but at the same time I cannot find a good place in the existing docs for it, so +1 to merge as is.

**Pull request merged**